iceoryx2/src/service/attribute.rs

```diff
@@ -98,26 +98,37 @@
 //! ```
 
 use core::ops::Deref;
+use iceoryx2_bb_container::{byte_string::FixedSizeByteString, vec::FixedSizeVec};
+use iceoryx2_bb_derive_macros::ZeroCopySend;
 use iceoryx2_bb_elementary::CallbackProgression;
 use serde::{Deserialize, Serialize};
 
+use crate::constants::{MAX_ATTRIBUTES, MAX_ATTRIBUTE_KEY_LENGTH, MAX_ATTRIBUTE_VALUE_LENGTH};
+
+type AttributeKeyString = FixedSizeByteString<MAX_ATTRIBUTE_KEY_LENGTH>;
+type AttributeValueString = FixedSizeByteString<MAX_ATTRIBUTE_VALUE_LENGTH>;
+
+type KeyStorage = FixedSizeVec<AttributeKeyString, MAX_ATTRIBUTES>;
+type AttributeStorage = FixedSizeVec<Attribute, MAX_ATTRIBUTES>;
+
 /// Represents a single service attribute (key-value) pair that can be defined when the service
 /// is being created.
-#[derive(Debug, Serialize, Deserialize, Eq, PartialEq, Clone, PartialOrd, Ord)]
+#[derive(Debug, Eq, PartialEq, Clone, PartialOrd, Ord, ZeroCopySend, Serialize, Deserialize)]
+#[repr(C)]
 pub struct Attribute {
-    key: String,
-    value: String,
+    key: AttributeKeyString,
+    value: AttributeValueString,
 }
 
 impl Attribute {
     /// Acquires the service attribute key
     pub fn key(&self) -> &str {
-        &self.key
+        self.key.as_str().unwrap()
```

Comment:
All of these `unwrap()` calls will be looked at more closely before this PR is merged. The solution for each case is however not immediately obvious.

Done like this as a "quick intermediate step" to get the foundations in place.

Reply:
Analysed all instances of `unwrap()` - those that are there should be fine to keep in. The failure mode is prevented by invariants in the structs (i.e. construction with `&str` which guarantees valid utf-8).